sawtooth/client.py

```diff
@@ -779,24 +779,34 @@ def get_transaction_status(self, txnid, timeout=30):
             request.get_method = lambda: 'HEAD'
             opener = urllib2.build_opener(self.proxy_handler)
             response = opener.open(request, timeout=timeout)
-            code = response.getcode()
-            response.close()
-            if code == 200:
-                return transaction.Status.committed
-            elif code == 302:
-                return transaction.Status.pending
-            else:
-                return transaction.Status.unknown
 
         except urllib2.HTTPError as err:
             LOGGER.error('peer operation on url %s failed with response: %d',
                          url, err.code)
+            if err.code == 302:
```

Comment:
When a transaction submitted but has not been committed, then get the status for this transaction, this could reproduce this code path.  As I know, when transaction exists but has not been committed, server returns error_response 302 for getting the status of the transaction. Maybe we should reconsider this scenario.